src/main/java/graphql/schema/idl/TypeInfo.java

```diff
@@ -139,6 +139,36 @@ public Type unwrapOneType() {
         return unwrapOne().getRawType();
     }
 
+    /**
+     * Gets the {@link TypeName} type name of a [Type], unwrapping any lists or non-null decorations
+     *
+     * @param type the Type
+     *
+     * @return the inner {@link TypeName} for this type
+     */
+    public static TypeName getTypeName(Type<?> type) {
+        while (!(type instanceof TypeName)) {
+            if (type instanceof NonNullType) {
+                type = ((NonNullType) type).getType();
+            }
+            if (type instanceof ListType) {
+                type = ((ListType) type).getType();
+            }
+        }
+        return (TypeName) type;
+    }
+
+    /**
+     * Gets the string type name of a [Type], unwrapping any lists or non-null decorations
+     *
+     * @param type the Type
+     *
+     * @return the inner string name for this type
+     */
+    public static String typeName(Type<?> type) {
+        return getTypeName(type).getName();
+    }
```

Comment:
a version that is just the string of the `TypeName` - easier to use in map lookups